### DISCUSSION TITLE
ROADMAP: V.2

### DISCUSSION THREAD

**Discussion by sannajammeh** [sannajammeh/tw-classed#155] [Announcements]
# tw-classed version 2.

TW Classed is going through some changes! Most will be non-breaking for most users, the non breaking changes will land in minor releases before Version 2.

## Planned minor changes - one release per change

- [ ] #153 
- [ ] #143 
- [ ] #144
- [ ] #98

## Planned breaking changes
- [ ] Deprecate `forwardRef` for React 19
- [ ] Set minimum React version to 19

**Comment by u3u** [comment]
<img width="768" alt="image" src="https://github.com/user-attachments/assets/8be041dd-953e-4d20-b2a9-b961fb7390a0">

The [`asChild` prop](https://react-twc.vercel.app/docs/guides/as-child-prop) and [`className` prop](https://react-twc.vercel.app/docs/guides/class-name-prop) of [`twc`](https://react-twc.vercel.app/) look good, but I prefer `classed`, which was actually created earlier than `twc` but didn't become popular. Perhaps it's because `twc` is a better name.

```tsx
import { twc } from 'react-twc'
 
const Title = twc.h2`text-2xl font-bold`;
 
// Strings
<Title className="uppercase" />
 
// Array
<Title className={[false, 0, "uppercase"]} />
 
// Objects
<Title className={{ uppercase: true }} />
 
// Nesting
<Title className={[{ uppercase: true }, "underline"]} />
```
